### PR TITLE
doc: change UOS/SOS to User VM/Service VM

### PR DESCRIPTION
**Pull request by dbkinder**
First pass at updating obsolete usage of "UOS" and "SOS"

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

### CODE REVIEW

**Comment by deb-intel** [comment]
I think we should probably look at it won of these files to make sure the change is ok unless @gvancuts does not think it's necessary.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by dbkinder** [comment]
@NanlinXie Where do we go to see why the build failed? The docs built without error on my local system.

**Comment by dbkinder** [comment]
@NanlinXie is there a way to have CI recheck this PR, or do I need to do a force push to cause the check to happen again when --ignore-whitespace is added to the "git am" command in the CI check?

File: doc/developer-guides/hld/hv-interrupt.rst [hunk 1]
```diff
@@ -63,7 +63,9 @@ to support this. The ACRN hypervisor also initializes all the interrupt
 related modules like IDT, PIC, IOAPIC, and LAPIC.
 
 HV does not own any host devices (except UART). All devices are by
-default assigned to the Service VM. Any interrupts received by Guest VM (Service VM or User VM) device drivers are virtual interrupts injected by HV (via vLAPIC).
+default assigned to the Service VM. Any interrupts received by Guest VM
```
Comment (gvancuts) [inline doc/developer-guides/hld/hv-interrupt.rst:66]: Using "Guest VM" here makes me feel we're only talking about User VMs, how about we don't specify it and more generically say "received by Virtual Machines (Service VM and User VM)"

**Comment by gvancuts** [comment]
@dongyaozu @NanlinXie I think it would also make sense with this change that we try as much as possible to use these terms for code changes/new development. I don't know whether we want to go as far as changing the function names that use `uos` or `sos` in them but we should at least use these new terms in code comments and git commits.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by leilu** [comment]
OK to verify

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by wenlingz** [comment]
OK to verify

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by leilu** [comment]
OK to verify

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**